src/Block/BaseMediaBlockService.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Block;
+
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use Sonata\AdminBundle\Form\Type\ModelListType;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Block\Service\AbstractBlockService;
+use Sonata\BlockBundle\Block\Service\EditableBlockService;
+use Sonata\BlockBundle\Form\Mapper\FormMapper;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\Doctrine\Model\ManagerInterface;
+use Sonata\MediaBundle\Admin\BaseMediaAdmin;
+use Sonata\MediaBundle\Model\MediaInterface;
+use Sonata\MediaBundle\Provider\Pool;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Form\FormBuilder;
+use Symfony\Component\HttpFoundation\Response;
+use Twig\Environment;
+
+/**
+ * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ */
+abstract class BaseMediaBlockService extends AbstractBlockService implements EditableBlockService
+{
+    /**
+     * @var BaseMediaAdmin
+     */
+    protected $mediaAdmin;
+
+    /**
+     * @var ManagerInterface
+     */
+    protected $mediaManager;
+
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    public function __construct(
+        Environment $twig,
+        ContainerInterface $container,
+        ManagerInterface $mediaManager
+    ) {
+        parent::__construct($twig);
+
+        $this->mediaManager = $mediaManager;
+        $this->container = $container;
+    }
+
+    /**
+     * @return Pool
+     */
+    public function getMediaPool()
```

Comment:
Can't these methods include a return type declaration?

Reply:
I just cherry-pick this PR from one of my branch, it will be done in other PR. I add issue for 4.0 milestone about close api.

Reply:
Its fine. IMO, then we can wait to have these methods with its corresponding declarations before considering it as RTM.

Reply:
Done.